app/assets/javascripts/oxalis/view/components/editable_text_icon.js

```diff
@@ -0,0 +1,62 @@
+// @flow
+
+import React from "react";
+import { Input, Icon } from "antd";
+
+type EditableTextIconPropType = {
+  icon: string,
+  onChange: Function,
+};
+
+class EditableTextIcon extends React.PureComponent {
+  props: EditableTextIconPropType;
+
+  state: {
+    isEditing: boolean,
+    value: string,
+  } = {
+    isEditing: false,
+    value: "",
+  };
+
+  handleInputChange = (event: SyntheticInputEvent) => {
+    this.setState({ value: event.target.value });
+  };
+
+  handleOnChange = () => {
+    const value = this.state.value;
+
+    if (value !== "") {
+      this.props.onChange(this.state.value);
+    }
+    this.setState({ isEditing: false, value: "" });
+  };
+
+  render() {
+    const iconStyle = { cursor: "pointer" };
+
+    if (this.state.isEditing) {
+      return (
+        <Input
+          value={this.state.value}
+          onChange={this.handleInputChange}
+          onPressEnter={this.handleOnChange}
+          onBlur={this.handleOnChange}
```

Comment:
It might be a bit confusing that `onPressEnter` and `onBlur `are `{this.handleOnChange}`, while `onChange` is not `{this.handleOnChange}` but `{this.handleInputChange}`. My point is that the name `onChange` might be missleading